### DISCUSSION TITLE
How can I let users provide their own keys for an engine?

### DISCUSSION THREAD

**Discussion by pe1uca** [searxng/searxng#3486] [Q&A] [answered]
In my private instance I'd like to implement a custom search engine for another service I run.  
The issue is that service is scoped by user, so right now that it's only me using my searxng instance is fine to hard code everything into [the code](https://docs.searxng.org/_modules/searx/engines/demo_online.html) or in [the engine's settings' api_key](https://docs.searxng.org/admin/settings/settings_engine.html)  
But if I want to share searxng's instance with friends and family they'll be able to only search into a single user of the other service.  

How can I configure searxng to require an api key or some credentials for an engine?  
I see there's a setting called `require_api_key`, but not sure what it does.

**Comment by Austin-Olacsi** [comment, marked as answer]
AFAIK there is no way for the end user to set an api key, you need to be the instance owner to do that.

**Reply by return42** [reply]
> How can I configure searxng to require an api key or some credentials for an engine?

I have never used or tested, but an alternative is to set a token, see [Private Engines (tokens)](https://docs.searxng.org/admin/settings/settings_engine.html#private-engines-tokens)
